File: src/game/creature/CreatureType.java
package game.creature;


public enum CreatureType {
	Player,
	Zombie,
	Turret,
	TurretZombie,
	ZombieTurret;
	
	public Creature instantiate() {
		switch(this) {
		case Player:
			return new Player();
		case Zombie:
			return new Zombie();
		case Turret:
			return new Turret();
		case TurretZombie:
			return new TurretZombie();
		case ZombieTurret:
			return new ZombieTurret();
		}
		return null;
	}
	
	public int getCode() {
		switch(this) {
		case Player:
			return 0;
		case Zombie:
			return 1;
		case Turret:
			return 2;
		case TurretZombie:
			return 3;
		case ZombieTurret:
			return 4;
		}
		return -1;
	}
	
	public static CreatureType fromCode(int code) {
		switch(code) {
		case 0:
			return Player;
		case 1:
			return Zombie;
		case 2:
			return Turret;
		case 3:
			return TurretZombie;
		case 4:
			return ZombieTurret;
		}
		return null;
	}
}

File: ts/game/creature/CreatureType.ts
import { Creature } from './Creature.js';
import { Player } from './Player.js';
import { Zombie } from './Zombie.js';
import { Turret } from './Turret.js';
import { TurretZombie } from './TurretZombie.js';
import { ZombieTurret } from './ZombieTurret.js';

export enum CreatureType {
  Player,
  Zombie,
  Turret,
  TurretZombie,
  ZombieTurret
}

export function instantiateCreature(t: CreatureType): Creature {
  switch (t) {
    case CreatureType.Player:
      return new Player();
    case CreatureType.Zombie:
      return new Zombie();
    case CreatureType.Turret:
      return new Turret();
    case CreatureType.TurretZombie:
      return new TurretZombie();
    case CreatureType.ZombieTurret:
      return new ZombieTurret();
  }
}
export function getCode(t: CreatureType): number {
  return t;
}

export function fromCode(code: number): CreatureType {
  return code as CreatureType;
}
